rs/sns/governance/canister/canister.rs

```diff
@@ -677,7 +677,16 @@ async fn mint_tokens(request: MintTokensRequest) -> MintTokensResponse {
 #[cfg(feature = "test")]
 #[update]
 fn advance_target_version(request: AdvanceTargetVersionRequest) -> AdvanceTargetVersionResponse {
+    use ic_sns_governance::pb::v1::upgrade_journal_entry;
+    governance_mut().push_to_upgrade_journal(upgrade_journal_entry::Event::TargetVersionSet(
```

Comment:
Let's instead add a function `advance_target_version` on SNS Governance that would be used by the production code (via proposals) and in testing (via this function); let's make it perform the required sanity checks for `request.target_version`.

```suggestion
let result = governance_mut().advance_target_version(request);
AdvanceTargetVersionResponse::from(result)
```

Reply:
Done!